lib/helpers/path/setDottedPath.js

```diff
@@ -6,6 +6,15 @@ module.exports = function setDottedPath(obj, path, val) {
     return;
   }
   const parts = path.split('.');
+
+  if (
```

Comment:
Use `lib/helpers/specialProperties.js` instead of listing these properties out explicitly. Like how `clone()` handles it: https://github.com/Automattic/mongoose/blob/6f951f3baadd3a752d3e04bccb26b2163dd3e2a8/lib/helpers/clone.js#L141-L143

Also, we should skip special properties rather than throw an error.

Reply:
improved.